trio/socket.py

```diff
@@ -210,3 +92,481 @@
             IP_BIND_ADDRESS_NO_PORT = 24
 
 del sys
+
+
+# The socket module exports a bunch of platform-specific constants. We want to
+# re-export them. Since the exact set of constants varies depending on Python
+# version, platform, the libc installed on the system where Python was built,
+# etc., we figure out which constants to re-export dynamically at runtime (see
+# below). But that confuses static analysis tools like jedi and mypy. So this
+# import statement statically lists every constant that *could* be
+# exported. It always fails at runtime, since no single Python build exports
+# all these constants, but it lets static analysis tools understand what's
+# going on. There's a test in test_exports.py to make sure that the list is
+# kept up to date.
+try:
```

Comment:
Maybe we should put this in an `if TYPE_CHECKING` block? But eh, I guess it does literally say "It always fails at runtime" and I don't know how this might change tools other than type checkers. Ig we can push this change off if you want!

Reply:
looks like `jedi` doesn't care, so switching it to be a `TYPE_CHECKING` block instead.